torch/csrc/cuda/python_nccl.cpp

```diff
@@ -374,7 +374,7 @@ static std::vector<at::Tensor> extract_tensors(PyObject* obj) {
           "expected Tensor at %d (got %s)", (int)i, Py_TYPE(item)->tp_name);
     }
     auto var = (THPVariable*)item;
-    list.emplace_back(var->cdata.data());
+    list.emplace_back(var->cdata);
```

Comment:
@teng-li @pietern Does it matter if this function can now potentially return requires-grad tensors instead of non-requires-grad tensors? The current code is using `var->cdata.data()`, and with the Variable/Tensor merge we want to clean up all `.data()` callsites and use `var->cdata` directly, if possible. Thanks!

Reply:
Based on discussion with Sam, it's okay to remove `.data()` here, because the call sites of `extract_tensors()` just call `tensor.data_ptr()` on the result (e.g. https://github.com/pytorch/pytorch/blob/master/torch/csrc/cuda/python_nccl.cpp#L206, https://github.com/pytorch/pytorch/blob/master/torch/csrc/cuda/python_nccl.cpp#L287, https://github.com/pytorch/pytorch/blob/master/torch/csrc/cuda/python_nccl.cpp#L349), which still works after removing `.data()` here.